src/update-feeds/check-feeds.ts

```diff
@@ -0,0 +1,164 @@
+              ?.onChainValue ?? {
+              value: ethers.BigNumber.from(1),
+              timestamp: ethers.BigNumber.from(1),
+            };
+            const signedData = getStoreDataPoint(beacon.beaconId) ?? {
+              timestamp: '1',
+              airnode: ethers.constants.AddressZero,
+              encodedValue: ethers.constants.HashZero,
+              templateId: ethers.constants.HashZero,
+              signature: ethers.constants.HashZero,
+            };
+            const offChainValue = {
+              timestamp: ethers.BigNumber.from(signedData.timestamp),
+              value: decodeBeaconValue(signedData.encodedValue)!,
+            };
+
+            const localDataIsStale =
+              signedData?.airnode === ethers.constants.AddressZero ||
+              offChainValue.timestamp.lt(onChainValue.timestamp) ||
+              offChainValue.timestamp.gt(Math.ceil(Date.now() / 1000 + 60 * 60));
+            const currentValue = offChainValue.timestamp.gt(onChainValue.timestamp) ? offChainValue : onChainValue;
+
+            return {
+              ...beacon,
+              currentValue,
+              localDataIsStale,
+              signedData,
+            };
+          }),
+        },
+      }))
+      // Filter out dapis that cannot be updated
+      .filter((dapi) => {
+        const newBeaconSetValue = calculateMedian(
+          dapi.decodedDataFeed.beacons.map((beacon) => beacon.currentValue.value)
+        );
+        const newBeaconSetTimestamp = calculateMedian(
+          dapi.decodedDataFeed.beacons.map((beacon) => beacon.currentValue.timestamp)
+        )!.toNumber();
+        const adjustedDeviationThresholdCoefficient = multiplyBigNumber(
+          dapi.updateParameters.deviationThresholdInPercentage,
+          deviationThresholdCoefficient
+        );
+
+        return checkUpdateConditions(
+          dapi.dataFeedValue.value,
+          dapi.dataFeedValue.timestamp,
+          newBeaconSetValue,
+          newBeaconSetTimestamp,
+          dapi.updateParameters.heartbeatInterval,
+          adjustedDeviationThresholdCoefficient
+        );
+      })
+      // Transform the batch and exclude beacons that cannot be updated
+      .map((dapiInfo) => ({
+        dapiInfo,
+        updatableBeacons: dapiInfo.decodedDataFeed.beacons
+          .filter(({ localDataIsStale }) => !localDataIsStale)
+          .map(({ beaconId, airnodeAddress, signedData, templateId }) => ({
+            beaconId,
+            airnodeAddress,
+            signedData,
+            templateId,
+          })),
+      }))
+      // Finally, filter out dapis that cannot be updated (eg. if we don't have signed data)
+      .filter((dapi) => dapi.updatableBeacons.length > 0)
+  );
+};
+
+interface OnChainValue {
+  beaconId: string;
+  onChainValue: { timestamp: ethers.BigNumber; value: ethers.BigNumber };
+}
+
+export const callAndParseMulticall = async (
```

Comment:
This is specific for reading dataFeeds. Can we rename it to something more explicit? E.g. `multicallBeaconValues`